docs/extras/integrations/chat/llama2_chat.ipynb

```diff
@@ -0,0 +1,220 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "90a1faf2",
+   "metadata": {},
+   "source": [
+    "# Llama2 Chat\n",
+    "\n",
+    "This notebook shows how `LLM`s, that are able to serve a Llama-2 chat models (like `LlamaCPP`, `HuggingFaceTextGenInference`, ...), can be wrapped into `Llama2Chat` so that they support the required [Llama-2 chat prompt format](https://huggingface.co/blog/llama2#how-to-prompt-llama-2) without further modification. \n",
+    "\n",
+    "`Llama2Chat` can be used in applications as chat model. It implements `BaseChatModel`, converts a list of chat messages into the [required chat prompt format](https://huggingface.co/blog/llama2#how-to-prompt-llama-2) and forwards the formatted prompt as `str` to the wrapped `LLM`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "4ee0837d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/martin/Applications/miniconda3/envs/extern-langchain/lib/python3.10/site-packages/deeplake/util/check_latest_version.py:32: UserWarning: A newer version of deeplake (3.6.14) is available. It's recommended that you update to the latest version using `pip install -U deeplake`.\n",
+      "  warnings.warn(\n"
+     ]
+    }
+   ],
+   "source": [
+    "from langchain.chains import LLMChain\n",
+    "from langchain.chat_models import Llama2Chat\n",
+    "from langchain.memory import ConversationBufferMemory\n",
+    "from langchain.prompts.chat import (\n",
+    "    ChatPromptTemplate,\n",
+    "    HumanMessagePromptTemplate,\n",
+    "    MessagesPlaceholder,\n",
+    ")\n",
+    "from langchain.schema import SystemMessage"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2ff99380",
+   "metadata": {},
+   "source": [
+    "This is shown with `HuggingFaceTextGenInference` as an example. A `HuggingFaceTextGenInference` LLM encapsulates access to a [text-generation-inference](https://github.com/huggingface/text-generation-inference) server. In the following example, the inference server hosts a [meta-llama/Llama-2-13b-chat-hf](https://huggingface.co/meta-llama/Llama-2-13b-chat-hf) model. It was started with \n",
```

Comment:
It would be nice to show this running w/ [llama2 locally](https://python.langchain.com/docs/use_cases/question_answering/how_to/local_retrieval_qa), as well.

If you don't have it locally, I can add.

Reply:
Good idea, I'll give this a try today or tomorrow, extend the notebook and update this PR.